### PR TITLE
[RF] Remove details of likelihood minimization from public interface

### PR DESCRIPTION
**Pull request by guitargeek**
* moves `MinuitFcnGrad`, `RooGradMinimizerFcn`, and the implementations
  of `LikelihoodWrapper` and `LikelihoodGradientWrapper` out of the
  public interfaces

* removing these public interfaces is unproblematic because they have
  not been released yet

* new factory methods added to `Likelihood(Gradient)Wrapper` so we can
  still create specific implementations for testing

* `clang-format` applied

### CODE REVIEW

File: roofit/roofitcore/src/TestStatistics/LikelihoodGradientWrapper.cxx [hunk 1]
```diff
@@ -13,6 +13,11 @@
 #include "RooFit/TestStatistics/LikelihoodGradientWrapper.h"
 #include "RooMinimizer.h"
 
+// including derived classes for factory method
+#ifdef BUILD_WITH_ROOFIT_MULTIPROCESS
```
Comment (lmoneta) [inline roofit/roofitcore/src/TestStatistics/LikelihoodGradientWrapper.cxx:17]: Where are these macro defined ?
Would not be better to have something as in other ROOT libraries defined in RConfigure.h and with a name like: 
R__HAS_ROOFIT_MULTIPROCESS

**Review by lmoneta — APPROVED** [review]
I agree with these improvements, it is better to hide these details from the public interface.
Thank you for the improvements!

File: roofit/roofitcore/src/TestStatistics/LikelihoodGradientWrapper.cxx (hunk 1 above)
Comment (guitargeek) [reply roofit/roofitcore/src/TestStatistics/LikelihoodGradientWrapper.cxx]: That's a good point! I have added a new commit with what you suggested.

Comment (guitargeek) [reply roofit/roofitcore/src/TestStatistics/LikelihoodGradientWrapper.cxx]: The macro existed already before this PR, by the way

**Comment by egpbos** [comment]
Bit late to the party, but just for the record: looks good to me. One thing I would consider, based on this, is to also hide away RooMinimizerFcn. Of course, this is impossible, because it is already there, but it's good for consistency probably to think about it for possible future versions.

**Pull request merged**